src/components/policy/policy_external/src/sql_pt_representation.cc

```diff
@@ -2660,11 +2660,18 @@ bool SQLPTRepresentation::InsertVehicleDataItem(
   }
 
   if (vehicle_data_item.params->is_initialized()) {
+    std::map<std::string, std::string> stored_vehicle_data_item_params;
     for (const auto& param : *(vehicle_data_item.params)) {
       if (!InsertVehicleDataItem(param)) {
         return false;
       }
 
+      if (stored_vehicle_data_item_params.end() !=
+          stored_vehicle_data_item_params.find(param.name)) {
```

Comment:
It doesn't seem like this would work for parameters which have multiple definitions, such as:
```
                        {
                            "name": "utcMonth",
                            "key": "OEM_REF_UTC_MONTH",
                            "type": "Integer",
                            "mandatory": false,
                            "minvalue": 1,
                            "maxvalue": 12,
                            "since": "5.0"
                        },
                        {
                            "name": "utcMonth",
                            "key": "OEM_REF_UTC_MONTH",
                            "type": "Integer",
                            "mandatory": true,
                            "minvalue": 1,
                            "maxvalue": 12,
                            "since": "2.0",
                            "until": "5.0"
                        },
```

Reply:
@jacobkeeler the point is it actually shouldn't. `InsertVehicleDataItem` will store both items from your example to `vehicle_data_item_definitions` table. 
But the `vehicle_data_item_parameters` table should contain only a single row with name and key of the parameter. Further, when retrieving items, using the name and key from `vehicle_data_item_parameters` table, both of these items will be retrieved.

Reply:
This was by design, to my knowledge. The correct schema would be selected based on the version of the mobile application. 

If we aren't going to be handling versioning for these parameters, then older definitions of parameters should be removed from the preloaded pt. Otherwise there is some ambiguity over which schema definition would be used. In this case, I suppose the best approach would be to use the latest definition, but use the original `since` value:
```
                        {
                            "name": "utcMonth",
                            "key": "OEM_REF_UTC_MONTH",
                            "type": "Integer",
                            "mandatory": false,
                            "minvalue": 1,
                            "maxvalue": 12,
                            "since": "2.0"
                        },
```

Reply:
@jacobkeeler Versioning is okay, and yes, you are right, this was by design. This fix is needed to make it work as it was by design. Now it duplicates results upon request of the item (as many times as you find same name in the table)

Reply:
@mkorniichuk 👍 Alright, looking at the structure of the tables, I can see what you're saying now